kasa/__init__.py

```diff
@@ -72,11 +69,54 @@
     "DeviceFamilyType",
 ]
 
+from . import iot
+
 deprecated_names = ["TPLinkSmartHomeProtocol"]
+deprecated_smart_devices = {
+    "SmartDevice": iot.IotDevice,
+    "SmartPlug": iot.IotPlug,
+    "SmartBulb": iot.IotBulb,
+    "SmartLightStrip": iot.IotLightStrip,
+    "SmartStrip": iot.IotStrip,
+    "SmartDimmer": iot.IotDimmer,
+    "SmartBulbPreset": BulbPreset,
+}
 
 
 def __getattr__(name):
     if name in deprecated_names:
         warn(f"{name} is deprecated", DeprecationWarning, stacklevel=1)
         return globals()[f"_deprecated_{name}"]
+    if name in deprecated_smart_devices:
+        new_class = deprecated_smart_devices[name]
+        package_name = ".".join(new_class.__module__.split(".")[:-1])
+        warn(
+            f"{name} is deprecated, use {new_class.__name__} "
```

Comment:
Maybe also mention, that the preferred way is to use use devicefactory/device.connect?
"Use X or class Y from package Z instead"?

Reply:
Done